### PR TITLE
Fix JVP rule for lax.pow()

### LINKED ISSUE #12033: The second order gradient of `lax.pow` is wrong
The second order gradient of `lax.pow` is wrong. For instance, the `fn` computes `x^y`, so `grad_fn` returns `(y*x^(y-1), ln(x)* x^y)`. When `x, y = 23.0, 0.0`, for the second element `ln(x) * x^y`, the gradient w.r.t `x` should be `1/x` instead of `23.0` computed by `jacrev` or `jacfwd`. 

```python
import jax
def grad_fn(x, y):
    def fn(x, y):
        return jax.numpy.sum(jax.lax.pow(x, y))
    return jax.grad(fn, (0, 1))(x, y)
x = jax.numpy.array([23.0])
y = jax.numpy.array([0.0])

print(jax.jacrev(grad_fn, (0, 1))(x, y))
print(jax.jacfwd(grad_fn, (0, 1))(x, y))
```

```
((DeviceArray([[0.]], dtype=float32), DeviceArray([[23.]], dtype=float32)), (DeviceArray([[0.04347826]], dtype=float32), DeviceArray([[9.831324]], dtype=float32)))
```

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #12033

### CODE REVIEW

File: jax/_src/lax/lax.py [hunk 1]
```diff
@@ -2001,8 +2001,7 @@ def _abs_jvp_rule(g, ans, x):
 pow_p = standard_naryop([_float | _complex, _float | _complex], 'pow')
 
 def _pow_jvp_lhs(g, ans, x, y):
-  jac = mul(y, pow(x, select(eq(y, _zeros(y)), _ones(y), sub(y, _ones(y)))))
-  return mul(g, jac)
+  return mul(g, mul(y, pow(x, sub(y, _ones(y)))))
```
Comment (jakevdp) [inline jax/_src/lax/lax.py:2004]: It's unclear to me why this rule ever included a select statement – @mattjj, @froystig, do you have any recollection why it was originally implemented this way?

Comment (jakevdp) [reply jax/_src/lax/lax.py:2004]: I found the PR that implemented the orginal logic: https://github.com/google/jax/pull/383

Comment (jakevdp) [reply jax/_src/lax/lax.py:2004]: Some git blame sleuthing by @mattjj found the origin of this select statement: https://github.com/HIPS/autograd/issues/116

Comment (mattjj) [reply jax/_src/lax/lax.py:2004]: Whoever wrote that line is an idiot!

**Comment by jakevdp** [comment]
I double-checked against the analog of the code in https://github.com/HIPS/autograd/issues/116, which inspired the introduction of this select statement. I get the same result before and after the change in this PR:
```python
import jax
import jax.numpy as jnp

def fun(x):
    return jnp.exp(-(x*x))
d1 = jax.grad(fun)
d2 = jax.grad(d1)
d3 = jax.grad(d2)
d4 = jax.grad(d3)
print(d1(0.), d2(0.), d3(0.), d4(0.))
# -0.0 -2.0 0.0 12.0

def fun(x):
    return jnp.exp(-(x**2))
d1 = jax.grad(fun)
d2 = jax.grad(d1)
d3 = jax.grad(d2)
d4 = jax.grad(d3)
print(d1(0.), d2(0.), d3(0.), d4(0.))
# -0.0 -2.0 0.0 12.0
```

**Comment by jakevdp** [comment]
So, it turns out there is one (untested) case that this PR changes:
```python
jax.grad(jax.lax.pow, 0)(0.0, 0.0)
```
On `main`, this returns `0.0`; with this PR, it returns `NaN`.

Chatting offline with @mattjj, we realized this is similar in spirit to the `sinc` zero issue in (#5054); the resolution there was to do a custom JVP rule using the maclaurin series (#5077), and I think the most rigorously-correct solution here will probably be similar.

**Review by mattjj — APPROVED** [review]
